src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -155,4 +149,33 @@ private static RequestAndRelatedRecords setRequestQueuePosition(
   private static boolean isTheSameRequester(RequestAndRelatedRecords it, Request that) {
     return Objects.equals(it.getUserId(), that.getUserId());
   }
+  
+  private static CompletableFuture<Result<Loan>> findOpenLoanWithRequest(RequestAndRelatedRecords requestAndRelatedRecords,
+      LoanRepository loanRepository) {
+
+    Request request = requestAndRelatedRecords.getRequest();
+
+    return loanRepository.findOpenLoanForRequest(request);
+  }
+
+  private static Result<RequestAndRelatedRecords> refuseWhenUserHasAlreadyBeenLoanedItem(RequestAndRelatedRecords requestAndRelatedRecords,
+      Result<Loan> loanResult) {
+
+    Request request = requestAndRelatedRecords.getRequest();
+
+    Loan loan = loanResult.value();
```

Comment:
Results can fail, meaning that the loan will be null, but not because it wasn't found.

Use of a result value should really be done via one of the methods of the class.

Reply:
Ok, I think the current state of the PR addresses this concern.